core/src/components/input/input.tsx

```diff
@@ -424,7 +424,11 @@ export class Input implements ComponentInterface {
    */
   private emitInputChange(event?: Event) {
     const { value } = this;
-    this.ionInput.emit({ value, event });
+
+    // Checks for both null and undefined values
+    const newValue = value == null ? value : value.toString();
```

Comment:
The type interface only allows `string | null`, no `undefined`, so should we return `null` instead of `value` on the left half? Or should the type be updated?

Reply:
Not sure I understand. The interface for `InputInputEventDetail` has `value?: string | null` where the `?` indicates the value can be `undefined`.

Reply:
Oh, you're right, I missed that 🤪